validation/src/main/java/io/serverlessworkflow/validation/state/StatesValidator.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.validation.state;
+
+import io.serverlessworkflow.api.Workflow;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.states.DefaultState.Type;
+import io.serverlessworkflow.api.validation.ValidationError;
+import java.util.Collection;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
+public final class StatesValidator {
+  private final Map<Type, StateValidator> stateValidators = new EnumMap<>(Type.class);
+
+  private final Collection<ValidationError> validationErrors;
+
+  public StatesValidator(Collection<ValidationError> validationErrors) {
+    this.validationErrors = validationErrors;
+
+    stateValidators.put(Type.EVENT, new EventStateValidator(validationErrors));
```

Comment:
Do we have to create the instance of validator for event type every we want to validate states?

Reply:
Actually, it will create only one instance of `StatesValidator` per `WorkflowValidatorImpl` instance.
See: https://github.com/hbelmiro/sdk-java/blob/cdf134a2d324310e7fa37f6e63823019a104d4d4/validation/src/main/java/io/serverlessworkflow/validation/WorkflowValidatorImpl.java#L54
That instance of `StatesValidator` has several instances of `StateValidator` (one per type) - I know I need to think about better names :)
In the end, only one instance of each class will be created.